ocaml/xapi/message_forwarding.ml

```diff
@@ -610,6 +610,14 @@ functor
           Ref.string_of vusb
       with _ -> "invalid"
 
+     let repository_uuid ~__context repository =
+       try
+         if Pool_role.is_master () then
+           Db.Repository.get_uuid __context repository
+         else
+           Ref.string_of repository
+       with _ -> "invalid"
```

Comment:
This does not seem like a good solution. It would be better to use an option value if this can be invalid.